### PR TITLE
Lucile fix user filter if found users is less than 10

### PR DESCRIPTION
**Pull request by LucileTech**
# Description
9.  (PRIORITY LOW) Aaron/Jae: Fix user filter if found users is less than 10 (WIP Lucile)
Filtering users displays the wrong number of users if number of users is less than 10
Admin/Owner login -> Other Links -> User Management -> search for users (ex. devadmin)
Notice how it says 1-10 when there's only 4 users

## Main changes explained:
- Changed the conditional rendering In UserTableFooter

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as admin user
4. go to dashboard→ Other Links -> User Management -> search for users (ex. devadmin)
5. verify that the filtering displays the right number with different choices of filter ( 10 / 25 / 50 )

## Screenshots or videos of changes:

Before 
Screenshot 2023-07-20 at 13 58 28

After
Screenshot 2023-07-20 at 13 52 44
Screenshot 2023-07-20 at 13 57 32

### CODE REVIEW

**Review by ramyaram2092 — DISMISSED** [review]
Works as expected. Approving the PR on the same note 

**Testing Video** 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/89441435/0a4698c1-bc4d-4070-913f-02d12209c5ff

**Comment by AnishPandita** [comment]
Hi @LucileTech the PR seems to work fine. Checked with 10, 25, 50 filter options.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111321945/8a7c93df-b4f5-44be-bf8c-af34ba9b0191

**Review by AaronPersaud — CHANGES_REQUESTED** [review]
Hi, I've tested it and it works as expected.

<b>Less than 10 users</b>

users

However, I believe it makes more sense to not remove the starting number. "Showing 11-20 of 20" is more intuitive and is the more common practice than "Showing 20 of 20"

<b>Before</b>

page2_users

<b>After</b>

image

- when the 11 is not included, this way sounds like there's 20 items displayed which is wrong

`Showing 1-10 of 4` should be changed to `Showing 1-4 of 4`

Another thing, which was already there before this fix, is that it says "Show" when "Showing" is grammatically correct.

**Review by iTvX — COMMENTED** [review]
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/e4119e17-923f-45b5-939d-943b12313d56

I have tested using "set final day", the number is exactly the same. Great work!

File: src/components/UserManagement/UserTableFooter.jsx [hunk 1]
```diff
@@ -91,8 +91,11 @@ const PageSummaryLabel = React.memo(props => {
       className="table-summary col-md-4 col-sm-4 col-xs-4 ember-view"
       style={{ marginBottom: '25px' }}
     >
-      {`Show ${(props.selectedPage - 1) * props.pageSize + 1} -
-    ${props.selectedPage * props.pageSize} of ${props.datacount}`}
+      {`Show `}
+      {props.datacount > props.selectedPage * props.pageSize
```
Comment (iTvX) [inline src/components/UserManagement/UserTableFooter.jsx:95]: Maybe it's more readability write like this:
``` Javascript
const displayedUsers = props.datacount > props.selectedPage * props.pageSize
    ? props.selectedPage * props.pageSize
    : props.datacount;
```

**Review by beblicarl — APPROVED** [review]

**Review by BaileyM96 — APPROVED** [review]
Good morning @LucileTech I have reviewed your changes and everything works great!
The only issue I faced was when I set it to show 25 users the page numbers in the bottom right corner are covered by the "back to top" button, I will include a screenshot to it down below.

## Video Walkthrough

[PR #1027 Fix user filter with less than 10.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108101478/fdcf104a-a3fa-48a5-aa50-87281357a868)

## Screenshots

As you can see the page numbers to switch to are covered by the back to top button on my screen, that issue is probably related to a past PR but I wanted to bring it to your attention

PR #1027 Back to top UI error

File: src/components/UserManagement/UserTableFooter.jsx [hunk 2]
```diff
@@ -91,8 +91,11 @@ const PageSummaryLabel = React.memo(props => {
       className="table-summary col-md-4 col-sm-4 col-xs-4 ember-view"
       style={{ marginBottom: '25px' }}
     >
-      {`Show ${(props.selectedPage - 1) * props.pageSize + 1} -
-    ${props.selectedPage * props.pageSize} of ${props.datacount}`}
+      {`Show `}
```
Comment (cvtqx) [inline src/components/UserManagement/UserTableFooter.jsx:94]:
```suggestion
     {`Showing `}
      {props.datacount <= 10 && `${props.datacount}`}
      {props.datacount > 10 && props.datacount > props.selectedPage * props.pageSize
        && `${(props.selectedPage - 1) * props.pageSize + 1} -
    ${props.selectedPage * props.pageSize}`}
      {` of ${props.datacount}`}
```

Comment (cvtqx) [reply src/components/UserManagement/UserTableFooter.jsx:94]: My suggestion on how to handle message depending on the number of users. Also I agree with a previous commenter that 'Showing' instead of 'Show' seems more correct.

**Review by cvtqx — CHANGES_REQUESTED** [review]
Left code suggestion to handle text depending on the number of users.

**Review by eduvarjaor — DISMISSED** [review]
I tested and everything it's correct. Great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86692035/32503816-150d-4ea8-abbb-471614934014

**Review by sxiong5 — COMMENTED** [review]
Hey, I tested the changes, and it worked as expected. But same as the last few comments, it's better to show `{start} - {end} of {totalNum}`. Consider the following code to make the code and UI page more readable.

```js
const PageSummaryLabel = React.memo(props => {
  const start = (props.selectedPage - 1) * props.pageSize + 1;
  const end = Math.min(props.datacount, props.selectedPage * props.pageSize);

  return (
    <div
      id="user_table_footer"
      className="table-summary col-md-4 col-sm-4 col-xs-4 ember-view"
      style={{ marginBottom: '25px' }}
    >
      Show {start} - {end} of {props.datacount}
    </div>
  );
});
```
Screenshot 2023-07-21 at 14 12 23
Screenshot 2023-07-21 at 14 06 35
Screenshot 2023-07-21 at 14 06 27

**Comment by mahamakhan** [comment]
Function mentioned works perfectly fine.

image

**Review by mahamakhan — APPROVED** [review]
Function mentioned works perfectly.

image

**Comment by Shreyj14** [comment]
Hi, @LucileTech, I checked this PR and it works as intended. 
**For 4 users**
Screenshot 2023-07-21 at 2 48 29 PM

**For 10 users:**
Screenshot 2023-07-21 at 2 49 01 PM

**Review by Shreyj14 — CHANGES_REQUESTED** [review]
As mentioned in the earlier comments, use something like ->
showing 1-4 of 4, 
showing 1-10 of 250, 
which will provide better understanding to users.

**Review by HectorAgudelo — DISMISSED** [review]
THis Pr works as expected. The numbers match users shown.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/37153730/8ac417b5-45e4-447f-b735-ca541f01c872

**Review by shiva2096 — COMMENTED** [review]
Hi @LucileTech 

Changes are working as expected, however some changes can be made to make the information more interpretable.
The info at the bottom left corner regarding {start}-{end} of {total} gives a better understanding


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/26687177/2b808eb3-e19d-41c4-9dab-347b3c7e32d0

**Review by vemanamadhu — CHANGES_REQUESTED** [review]
The functionality works well.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13665284/0e35ddf4-30c8-4710-af64-2d5b9678c9ae


You could create a variable and assign the value to it and then use that variable later. I feel like that makes the code more readable.
const NumberUsers = props.datacount > props.selectedPage * props.pageSize
    ? props.selectedPage * props.pageSize
    : props.datacount;

**Review by yaow62023 — APPROVED** [review]
This PR works as expected. For different numbers of search results, the correct number can be displayed, such as "Show 10 of 11" when the filter is set to 10 results, and all results are displayed when it is set to 25, such as "Show 11 of 11".

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/af48dad0-211a-44fe-a22d-2f2a44433211

**Review by yurisokolovicz — APPROVED** [review]
The new feature is working fine. Approving PR. Good job.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/121908988/34d7b941-8990-4bd6-88a1-04ac9f09be93

**Review by n0vay — COMMENTED** [review]
Good Job,
The code works as intended, its shows the number correctly if the users are less than 10.
For more than 10 users it can say showing 10-20.
image

Also if we see 50 users, the back to top button comes over next button.
image

**Review by winghojackyli — APPROVED** [review]
I tested the PR and it worked as expected. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/9e8f1935-9eb9-4296-aae3-a76fc35e0dff

**Comment by mounica-dingari-by** [comment]
Hi @LucileTech , I've tested the PR and it works as expected. 

Screenshot from 2023-07-22 22-22-46

**Comment by one-community** [comment]
> `Showing 1-10 of 4` should be changed to `Showing 1-4 of 4`
> 
> Another thing, which was already there before this fix, is that it says "Show" when "Showing" is grammatically correct.
Yes to all this

File: src/components/UserManagement/UserTableFooter.jsx (hunk 2 above)
Comment (LucileTech) [reply src/components/UserManagement/UserTableFooter.jsx]: thank you

File: src/components/UserManagement/UserTableFooter.jsx (hunk 1 above)
Comment (LucileTech) [reply src/components/UserManagement/UserTableFooter.jsx]: thank you

**Review by AaronPersaud — DISMISSED** [review]
Good afternoon, I've tested it and it looks good!

<b>Less than 10 users</b>

image

- uses the format "Showing 1 to `{datacount}` of `{datacount}`" when < 10 users

<b>Page 2</b>

image

- works as normal

Nice touch with the refactor. The variables and conditional are a lot easier to understand now.

**Review by EdwinLauM — APPROVED** [review]
Hey @LucileTech! Functionality works as intended! Good job! 

Less than 10 users:

Screen Shot 2023-07-27 at 1 49 18 AM

Filtering 50 users:

Screen Shot 2023-07-27 at 1 49 58 AM

**Review by cvtqx — DISMISSED** [review]
The changes look great! the code also looks clear and concise. Well done!

1. When found less than 10 users:
6 users

2. When found more than 10 users + filter set to display 25:
allFoundUsers

3. When found more than 10 users  - paginated view:

Page1:
page1

Page2:
page2

**Comment by n0vay** [comment]
Good Job on the code, works perfectly.

**Comment by n0vay** [comment]
I am getting a memory leak error.

**Comment by Lin1404** [comment]
Hi @LucileTech,

I tested the changes, and it worked as expected. 

The paginated view for less than 10:
image


The paginated view for more than the number filter set to display:
image

**Review by Lin1404 — DISMISSED** [review]

**Comment by Indujaks** [comment]
I tested your changes for all the possible cases, it works perfectly fine when the **user is found**!
Screenshot 2023-07-28 at 12 08 13 PM

**But when the user is not found it says 1-0** 
Screenshot 2023-07-28 at 12 08 23 PM
of 0, can you update it to say 0 instead

**Review by Indujaks — DISMISSED** [review]
I tested your changes for all the possible cases, it works perfectly fine when the user is found!
Screenshot 2023-07-28 at 12 08 13 PM


But when the user is not found it says 1-0 of 0, can you update it to say 0 instead
Screenshot 2023-07-28 at 12 08 23 PM

**Review by zuhang2000 — CHANGES_REQUESTED** [review]
I have reviewed your changes and did test on it. It's working great in many edge cases! 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123051008/4f0a3166-186b-445e-aabb-53bfc39dc565

Just one small problem as @Indujaks also points out earlier. I do notice this "1-0" when no user can be found. Will be perfect if it is solved!
Screen Shot 2023-07-29 at 1 11 20 AM

**Review by xaanders — APPROVED** [review]
Hi @LucileTech! I've tested you PR. It works perfectly.

[scrnli_8_2_2023_9-29-26 AM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108314277/1b470c78-b5ae-499f-9a97-0c2524f692a1)

**Review by TheDescendant39 — APPROVED** [review]
Hi @LucileTech , I tested your PR to check all the cases and it worked well. Good job.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47534216/ad9e8574-737d-4a54-aa44-2735a50983d6

**Review by Abiddy — APPROVED** [review]
LGTM!

Screenshot from 2023-08-05 23-20-58
Screenshot from 2023-08-05 23-20-49

**Review by EdwinLauM — APPROVED** [review]
Hey @LucileTech. Works as intended! Good job! 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/de22cbb4-fe6f-47e2-ba65-17f677d96830

**Review by KurtisIvey — APPROVED** [review]
Works perfectly under multiple queries. Can't say the same thing for that back button that kept getting in the way though lol.

Great job. Code looks good as well
Screenshot 2023-08-10 at 2 56 45 PM

**Comment by AuroraLovesWhales** [comment]
Hi @LucileTech. I have tested your PR and the user filter works as expected. Great work! It's not related to this PR, but when I set the filter to 25 or 50 and scroll to the bottom, the "Back to Top" button gets in the way of the changing page buttons. Here is my testing:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/8d8a4e82-4634-432b-be7f-d2573b161cee

**Review by EvianTan — APPROVED** [review]

**Review by EdwinLauM — APPROVED** [review]
Works as expected! Approving again! 

10 user filter:

PR #1027

25 user filter:

Screen Shot 2023-08-11 at 2 22 10 AM

50 user filter:

Screen Shot 2023-08-11 at 2 22 22 AM
